### PR TITLE
#155889952 Create java classes for resources

### PR DESCRIPTION
**Pull request by leonardnjura**
#### What does this PR do?
- Creates java model classes for use with retrofit

#### Description of Task to be completed?
- Create model package
- Add model classes

#### How should this be manually tested?
* Clone this repo: 
`$ git clone -b ft-java-classes-for-resources-155889952 https://github.com/leonardnjura/github-users`
* Open cloned directory `github-users` in Android Studio
* Build with Gradle
* Navigate to model directory:
`app/src/main/java/com/example/githubusers/model/`
* Optionally run app in emulator device or actual connected device i.e. a physical Android phone or tablet

#### What are the relevant pivotal tracker stories?
[#155889952](https://www.pivotaltracker.com/story/show/155889952)

#### Screenshots
None

### CODE REVIEW

File: app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.example.githubusers.model;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+
+public class GithubUsersResponse {
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java:8]: Being a model class, this class needs a `constructor` method

**Review by johngorithm — CHANGES_REQUESTED** [review]

**Comment by johngorithm** [comment]
screen shot 2019-03-05 at 6 28 50 pm

The above section of your PR description seemed a bit confusing, kindly ensure to keep it simple and in a step by step fashion

File: app/src/main/java/com/example/githubusers/model/GithubUsers.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    @SerializedName("last_name")
+    private String lastName;
+
+    @SerializedName("public_repos")
+    private int publicRepos;
+
+    @SerializedName("bio")
+    private String bio;
+
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/model/GithubUsers.java]: I believe this model class is to hold data for a single GitHub user in the `Github Response` for users. Therefore, be mindful for the available fields in the API Response and also what your Design `Mockup` requires of which I believe is just `username` and `imageUrl` for each user.

File: app/src/main/java/com/example/githubusers/model/GithubUsers.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package com.example.githubusers.model;
+
+import com.google.gson.annotations.SerializedName;
+
+
+public class GithubUsers {
+
+    @SerializedName("username")
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/model/GithubUsers.java]: Ensure that the names passed as an argument to the `@SerializedName` notation conform to the fields in the API response. I believe the above is denoted as `login` in the API Response.

File: app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    @SerializedName("result")
+    private List<GithubUsers> result;
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java]: This field `result` should describe what it contains

File: app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.example.githubusers.model;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+
+public class GithubUsersResponse {
+
+    @SerializedName("result")
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java]: I believe this(`"result"`) is not the field containing the list of users coming from the `API` kindly ensure the correct value from the API Response

**Review by johngorithm — CHANGES_REQUESTED** [review]
Good Job on your Learning. Kindly respond the Reviews below

File: app/src/main/java/com/example/githubusers/model/GithubUsers.java (hunk 2 above)
Comment (leonardnjura) [reply app/src/main/java/com/example/githubusers/model/GithubUsers.java]: Sure I will refer to `https://api.github.com/search/users?q=+language:java+location:nairobi&per_page=100` for list view and `https://api.github.com/users/{username}` for detail view

File: app/src/main/java/com/example/githubusers/model/GithubUsers.java (hunk 3 above)
Comment (leonardnjura) [reply app/src/main/java/com/example/githubusers/model/GithubUsers.java]: I fixed:
```
@SerializedName("login")
    private String login;
```

File: app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java (hunk 4 above)
Comment (leonardnjura) [reply app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java]: Totally agree. I found it is supposed to be `items`

File: app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+
+public class GithubUsersResponse {
+
+    @SerializedName("items")
+    private List<GithubUsers> items;
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java]: Make this field describe what it contains.

**Review by johngorithm — CHANGES_REQUESTED** [review]
Good Job 👍 . Just a minor change

File: app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java (hunk 6 above)
Comment (leonardnjura) [reply app/src/main/java/com/example/githubusers/model/GithubUsersResponse.java]: Okay

**Review by johngorithm — APPROVED** [review]
LGTM 👍 

@leonardnjura I am merging this.

**Pull request merged**